### ISSUE TITLE
Invoice Item discountable defaults to true

### ISSUE THREAD

**Issue by uorji3** [stripe/stripe-go#329]
According to the Stripe documentation, the `discountable` argument for invoice item 
> Defaults to false for prorations or negative invoice items, and true for all other invoice items.

For my project, I actually want a Stripe coupon to only discount the plan and not the invoice items. However, setting `discountable` to false is ignored here: https://github.com/stripe/stripe-go/blob/master/invoiceitem/client.go#L41 . I will replicate here for convenience:

```
	if params.Discountable {
		body.Add("discountable", strconv.FormatBool(true))
	}
```

The `discountable=false` argument is never added to the request to Stripe and it will be implied as true, which is not I wanted.

I propose we change the logic to at least:
```
	if (params.Amount > 0 && !params.Discountable) || (params.Amount < 0 && params.Discountable) {
		body.Add("discountable", strconv.FormatBool(params.Discountable))
	}
```
 or

```
	if (params.Amount > 0) != params.Discountable {
		body.Add("discountable", strconv.FormatBool(params.Discountable))
	}
```
This change will add the discountable param to the request if it is different from the default behavior.

**Comment by brandur** [comment]
Hi @uorji3, thanks for reporting and sorry about the trouble!

It's a little unusual, but to handle this sort of explicit false in booleans, we normally introduce a second parameter prefixed with `No*`. So for example, setting a false for `Discountable` would look like `NoDiscountable: true`.

I've added a fix over in #330.

**Comment by uorji3** [comment]
Makes sense with Golang's default zero values. Thanks for the quick turnaround.
